lib/spack/llnl/util/filesystem.py

```diff
@@ -1097,15 +1097,26 @@ def visit_directory_tree(root, visitor, rel_path='', depth=0):
     for f in dir_entries:
         if sys.version_info >= (3, 5, 0):
             rel_child = os.path.join(rel_path, f.name)
-            islink, isdir = f.is_symlink(), f.is_dir()
+            islink = f.is_symlink()
+            # Windows issue with attempting to read
+            # symlinked archive as dir results in too
+            # many reparse attempts and raises
+            # a CANT_RESOLVE_FILENAME error
+            # See https://bugs.python.org/msg357725
+            # for more context
+            try:
+                isdir = f.is_dir()
+            except OSError:
+                isdir = False
```

Comment:
Thanks for updating the comment! I'm worried this is too broad of a catch: `f.is_symlink` is called first, so would probably trigger most errors (e.g. permissions errors) befor `.is_dir` could, but I still think it might be worth taking care to check the error code (if possible) and/or `raise`ing on non-Windows systems. Does that seem reasonable?

Reply:
raise-ing on non Windows makes sense, and I'll dig into whether or not we can get more specificity on the Windows side.